django/contrib/staticfiles/management/commands/collectstatic.py

```diff
@@ -173,22 +173,27 @@ def handle(self, **options):
         if self.is_local_storage() and self.storage.location:
             destination_path = self.storage.location
             message.append(':\n\n    %s\n\n' % destination_path)
+            destination_exists = (
+                self.storage.exists(destination_path) and
+                any(self.storage.listdir(destination_path))
+            )
         else:
             destination_path = None
             message.append('.\n\n')
+            destination_exists = False
```

Comment:
I think the warning should happen in this case, since we didn't check if files exist or not, it's better to be safe.